### ISSUE TITLE
PostgreSQL 12 support

### ISSUE THREAD

**Issue by devrimgunduz** [timescale/timescaledb#1456]
Hi,
Timescaledb 1.4.2 does not compile against v12. Can you please release a new version? v12 is out next week.

Regards, Devrim

**Comment by bboule** [comment]
We are currently working on this, and will be in a better position to set timelines once we get a little further into the process, I will update this thread when we are ready to set timeline expectations.

**Comment by devrimgunduz** [comment]
Thanks!

**Comment by bluespidernav** [comment]
On Windows 10 64 bit -- I note that TimescaleDB also does not work with PostreSQL version 11.5.2 (well not straight out of the box) but worked fine with 11.0.3. The timescale extension DLL does not load as it had dependencies on the OpenSSL DLL's ssleay32.dll and libeay32.dll which were present in earlier PostgreSQL version but in 11.5 were replaced with the newer libcrypt/libssl. As a result two DLL's required by timescale were missing and as a workaround one could grab compatible versions and place them in the lib folder once you had figured out what was wrong. How horrible is that? Secondly for version 12 we find PostGIS is missing in stack builder so if you want both TimescaleDB and PostGIS then best to forget version 12 of PostgreSQL until that gets resolved. It seems right now the only available installer for 11 is 11.5 (older versions are not available on the EDB site for some unknown reason) so it seems you have to use that one and deal with those missing dlls. Is TimescaleDB ever going to be available in stack builder? Would be nice if it could be!

**Comment by fvannee** [comment]
I see milestone 1.6 was added to this issue - would you have any rough estimate when we could expect this release?

**Comment by bboule** [comment]
@fvannee  we do not yet we are still in the planning stages but we will be sure to update this thread as soon as the timeline becomes clear

**Comment by fvannee** [comment]
@bboule ok thank you for the update. let me know if a preliminary version would need to be tested at some point. i'd be happy to help.

**Comment by nailgilaziev** [comment]
Hello!
Usually after how many months support for the new version of postgresql is added?

**Comment by extesy** [comment]
And in the meantime PostgreSQL 12.1 is out: https://www.postgresql.org/docs/12/release-12-1.html

**Comment by jesperpedersen** [comment]
Minor versions doesn't change the API/SPI starting from PostgreSQL 10 - see https://www.postgresql.org/support/versioning/

**Comment by damoclark** [comment]
Hey @bboule 

> We are currently working on this, and will be in a better position to set timelines once we get a little further into the process, I will update this thread when we are ready to set timeline expectations.

Just wondering how your team is progressing with this?  Any updates?

Regards,
Damien.

**Comment by k-rus** [comment]
> > We are currently working on this, and will be in a better position to set timelines once we get a little further into the process, I will update this thread when we are ready to set timeline expectations.
> 
> Just wondering how your team is progressing with this? Any updates?

Current plan is to release at the beginning of next year as part of 1.7 release. I guess it will be more clear after all the holidays.

**Comment by apcmaff** [comment]
Assuming support will include 12.1+ ...by beginning of next year, is that Jan? Feb? or by end of Q1?  Thanks!

Best, 
Chris

**Comment by cevian** [comment]
The current plan of release is end-of-Jan beginning-of-Feb

**Comment by fvannee** [comment]
Thanks for the update @cevian

**Comment by webnard** [comment]
@cevian Are there any specific tasks that the community can do?

**Comment by zplzpl** [comment]
about the current progress?

**Comment by Ruizhijian** [comment]
> 
> 
> about the current progress?

Stop being Chinese everywhere. Not every one like you can make a 1000beds hospital in 7 days.

**Comment by devrimgunduz** [comment]
> > about the current progress?
> 
> Stop being Chinese everywhere. Not every one like you can make a 1000beds hospital in 7 days.

This is *very* rude and inappropriate.

v12 has been released a few months ago, and this extension is currently blocking updates. v12 has entered beta at the end of May 2020, and you are still talking in this tone?

Weird.

**Comment by akulkarni** [comment]
@Ruizhijian - This type of language is not appropriate here.

@devrimgunduz - I'm sorry about this. This individual is not associated with Timescale. But rest assured the team _is_ hard at work getting v12 support out the door!

**Comment by mfreed** [comment]
**Please note:**  We decided to delete one comment posted to this issue within the past 24 hours (and some follow-up responses).  This is a first for us, to my knowledge.

The comment, made by somebody who has no association with Timescale, was wildly inappropriate and divisive.  This type of sentiment has no place in our community, and we will not accept it.

As a side note to the broader discussion, the team _is_ hard at work getting v12 support out the door.  You can follow along with [the large PR here](https://github.com/timescale/timescaledb/pull/1525) (currently at 55 commits).

**Comment by vivekanandsh** [comment]
TimescaleDB Team, 
We are looking for **TimescaleDB version compatible to PostgreSQL-12**.  As mentioned before TimescaleDB for PG12 will be ready around Feb 2020. If beta version or version ready please help to share.  
If **NOT ready** let us know availability date. If available by Feb 2020 will be great. 
Waiting for response. 

Regards,
Vivek

**Comment by mfreed** [comment]
@vivekanandsh We are currently tracking to a March release date.

**Comment by fvannee** [comment]
@mfreed  What is Timescale's stance in general on supporting new versions of Postgres?
I love using TimescaleDb, but I found it usually causes delays in adopting newer Postgres versions. For example, PG11 was also only properly supported several months after it was released (version 1.2.0, end of January).
I understand it's quite a large undertaking to adopt all the changes in a new Postgres release, which is something that takes time. However, some other (also rather large and complex) extensions are often quite a bit faster in supporting new versions. Either they try to keep up-to-date with Postgres master during development, or they start their work to support new versions already when PG brings out the first early Beta release. I get the idea, but please correct me if I'm wrong, that Timescale's policy is to start working on support for a new version only after the final release of Postgres is released.

**Comment by mfreed** [comment]
@fvannee We don't have a hard-and-fast rule here, and scheduling engineering resources early on PG12 was balanced against other development priorities.

In this case, the main things we focused on in 2019, rather than prioritizing PG12, was multi-node, compression, and continuous aggregates.  I understand that folks would have liked PG12 support out earlier, but looking back, I'm not sure we would have made different decisions at the time.  I also think those capabilities are really valued by the community.

We'll try to give PG13 a little more priority up front, but again it's always a balancing act.

I would also say that if the community would like to contribute by helping us track PG13 master more closely, this is really some place that we'd very much welcome the help, and might be a "right-sized" project for some external contributors?  Thoughts?

**Comment by vivekanandsh** [comment]
mfreed ,
I love using TimeScaleDB. Thanks for your engineering effort.  I do agree and good if timescaledb latest version released  after ( lesser time gap) major versions of PostgreSQL. As requested before i held my release to production as i am waiting for TimeScaleDB version compatible to PG12. As you mentioned tentative planned date  in March 2020. Good if get TimeScaleDB beta for PG12 by  first/second week of March 2020.

**Comment by mfreed** [comment]
For folks that are interested in possibly contributing to the continuous PG13 integration, please shoot me at email?  mike (at) timescale com

**Comment by gajus** [comment]
> @vivekanandsh We are currently tracking to a March release date.

Is this still the plan?

**Comment by bboule** [comment]
Yes at this point we are still tracking to March release.

**Comment by vivekanandsh** [comment]
@bbouble
Thanks we are waiting for March release for PG12.   Any beta release available help share. We will verify for PG12.

**Comment by tienlx93** [comment]
Hi,

I have multiple cluster version of postgresql installed
Any idea how to install timescaledb for specific version?
```
/usr/lib/postgresql/12/bin/postgres
/usr/lib/postgresql/10/bin/postgres
/usr/lib/postgresql/11/bin/postgres
/usr/lib/postgresql/9.5/bin/postgres
```
Error:
```
sudo timescaledb-tune
exit: unknown major PG version: PostgreSQL 12.2 (Ubuntu 12.2-2.pgdg16.04+1)
```

**Comment by vivekanandsh** [comment]
@mfreed ,
Any update on timescaledb for PpstgresSQL 12 version. We are waiting for this release.

**Comment by vivekanandsh** [comment]
TimescalDB Team,
Support for PostgresSQL 12 version is till tracking to March release ?

**Comment by R-Ali** [comment]
I hope you guys are making progress :)

**Comment by brolly759** [comment]
At this point I expect PG13 support before we get PG12 support :)

**Comment by vivekanandsh** [comment]
TimeScaleDB Team,

Any update on release plan for  PostgresSQL 12 version. Earlier planned for March 2020. Please let us know urgently.   If postponed/delayed also please update.

**Comment by pehlert** [comment]
Guys instead of spamming everyone's email accounts, why don't you just wait for an update? I'm sure the TimescaleDB team will update when they have something to share. 

Also, this is an open source project so if their pace is not fast enough for you, you can always create a PR yourselves.

**Comment by erimatnor** [comment]
We're currently doing internal testing of TimescaleDB with PostgreSQL 12 support. PostgreSQL 12 turned out to be a bigger task than anticipated due to a large number of internal changes in PostgreSQL, in particular related to partitioning and the introduction of table access methods. 

We should have a further update on this in the very near future.

**Comment by wrobell** [comment]
@brolly759. @vivekanandsh, @pehlert According to the latest newsletter, PostgreSQL 12 will be supported by TimescaleDB 1.7. See also https://github.com/timescale/timescaledb/milestone/39.

**Comment by vivekanandsh** [comment]
TimescaleDB  Team,
Many thanks will check this TimescaleDB 1.7 version.

**Comment by derekjwilliams** [comment]
Thanks so much for the huge effort to get timescale support for postgres 12.

**Comment by vivekanandsh** [comment]
From where to download **TimescaleDB 1.7 version** (for postgres 12)  binaries.

**Comment by mskarbek** [comment]
@vivekanandsh from the download page when it finally will be released. Stop nagging and wait for a proper release announcement.

**Comment by wrobell** [comment]
@vivekanandsh Please, monitor the progress at the [1.7 milestone page](https://github.com/timescale/timescaledb/milestone/39). There are still 8 tickets to be closed.

**Comment by zdm** [comment]
Who needs this postgres 12 at all? 
Let's freeze on 11 version - it can perform sql queries.

**Comment by mfreed** [comment]
Hey folks, just flagging that our PR was submitted against master for PG12 support, and it should be merged soon.

69 commits and 282 files changed!

https://github.com/timescale/timescaledb/pull/1807

cc: @devrimgunduz
